packages/app-form-builder/src/admin/plugins/formDetails/formSubmissions/FormSubmissionsList/useSubmissions.ts

```diff
@@ -27,47 +23,46 @@ interface Reducer {
 export const useSubmissions = (form: Pick<FbFormModel, "id">) => {
     const [state, setState] = useReducer<Reducer>((prev, next) => ({ ...prev, ...next }), {
         loading: false,
-        page: 0,
-        cursors: {
-            0: null
-        },
         exportInProgress: false,
-        submissions: [],
         sort: ["createdOn_DESC"]
     });
 
-    const client = useApolloClient();
-
     const { showSnackbar } = useSnackbar();
 
-    const loadSubmissions = async (fetchPolicy: FetchPolicy = "cache-first") => {
-        setState({ loading: true });
+    const listQuery = useQuery(LIST_FORM_SUBMISSIONS, {
+        variables: { form: form.id, sort: state.sort, limit: 20 }
+    });
 
-        const { data: res } = await client.query<
-            ListFormSubmissionsQueryResponse,
-            ListFormSubmissionsQueryVariables
-        >({
-            query: LIST_FORM_SUBMISSIONS,
-            variables: { form: form.id, sort: state.sort, after: state.cursors[state.page] },
-            fetchPolicy
-        });
+    const loadMoreOnScroll = useCallback(
+        debounce(scrollFrame => {
+            if (scrollFrame.top > 0.9) {
```

Comment:
We should also not be doing anything if `loading: true` (despite the fact that the function is debounced).